app/src/main/java/io/github/hidroh/materialistic/WebFragment.java

```diff
@@ -404,17 +434,21 @@ public void onProgressChanged(android.webkit.WebView view, int newProgress) {
             }
         });
         mWebView.setDownloadListener((url, userAgent, contentDisposition, mimetype, contentLength) -> {
-            if (getActivity() == null) {
-                return;
-            }
-            final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-            if (intent.resolveActivity(getActivity().getPackageManager()) == null) {
-                return;
+            if (isPdfRenderingSupported() && mimetype.equals(PDF_MIME_TYPE)) {
+                downloadFileAndRenderPdf();
+            } else {
+                if (getActivity() == null) {
+                    return;
+                }
```

Comment:
This early termination should still be top of the callback, without context (activity here) we can't really do nothing else.

Reply:
Right